builder/vsphere/supervisor/step_watch_source.go

```diff
@@ -49,6 +49,7 @@ type StepWatchSource struct {
 
 	SourceName, Namespace string
 	KubeWatchClient       client.WithWatch
+	timeoutCh             <-chan time.Time
```

Comment:
I took a little time thinking about this give the provided info on Slack and in this PR, and think we can remove this entirely and change up the code a bit. I'll comment further down in the code with the proposed changes.


```suggestion
```